### PR TITLE
[OCPCLOUD-1575] Add section to API conventions to introduce motivation

### PR DESCRIPTION
**Pull request by JoelSpeed**
When asking for changes to API designs, I think it's important that the people we are working with understand what we are trying to achieve with API reviews and why they are important. 

I've made a start at this trying to assert that we are looking for consistency across the board of the API design, so that customers have a familiar look and feel to make the user experience better.
I've also, as this comes up a lot, added a note that we don't repeat the sins of our past, and, tried to expand on why we do that.

### CODE REVIEW

File: dev-guide/api-conventions.md [hunk 1]
```diff
@@ -255,3 +280,18 @@ Following this convention has the disadvantage that it deviates from what users
...
+
+### My proposed design looks like an existing API we have, why am I being told that it must be changed?
+
+There are a few reasons why an API reviewer might want you to change the proposed design of your API addition.
+
+When the API looks similar to an existing API, there are a couple of important things to bear in mind.
+
+Firstly, not all APIs in OpenShift have been through the API review process, therefore, especially early
+in the OpenShift 4 lifecycle, many APIs were shipped that were not compliant with the conventions.
+If you API looks like an existing API that is not compliant, it must be updated to be compliant.
```
Comment (lobziik) [inline dev-guide/api-conventions.md:294]: For me it's not quite clear what must be updated. Can this sentence be more specific?

Comment (JoelSpeed) [reply dev-guide/api-conventions.md]: Updated, is it clearer with the new wording?

**Review by elmiko — COMMENTED** [review]
i think this is a nice improvement
/lgtm

**Comment by dhellmann** [comment]
/approve
/hold

This provides good background for enhancement authors. I'm inclined to go ahead and merge it, since we're about to enter a period where I would expect more enhancements to be proposed. I could also see waiting for @deads2k or other API reviewers to take a look, though, so I'll drop a hold on it and you can decide whether to remove it.

File: dev-guide/api-conventions.md [hunk 2]
```diff
@@ -17,6 +17,34 @@ status: informational
...
+we must have a centralised set of conventions to ensure that the look and feel of our APIs is consistent
+across the board.
+By having consistent API design, our end users will feel familiar with the API no matter which of our APIs
+they are working with.
+
+API review also plays an important part in making sure that the APIs we release are of high quality and,
+where possible, consider the future needs of the API and the possible expansions we may make.
+
+As OpenShift APIs are supported immediately once they are merged, API reviews play an important part in
+making sure the API is correct (it has the sufficient validation, documentation and shape) and
```
Comment (Miciah) [inline dev-guide/api-conventions.md:38]:
```suggestion
making sure the API is correct (it has sufficient validation and documentation and a logical shape) and
```

**Review by Miciah — COMMENTED** [review]
A couple small suggestions but looks good overall.

**Pull request merged**